### PR TITLE
fix: use custom int parsing over flask int parsing in sqllab queries endpoint

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

Flask returns a 404 when a route doesn't match it's signature. `/superset/queries/1588736634550.6` was returning 404 due to the `<int:last_updated_ms>` syntax since the requested endpoint contains a float. 

This PR reverts the typing logic introduced in https://github.com/apache/incubator-superset/pull/9939/files#diff-ac978615a4f22c4fad4d01f39e1d4595L2646 in favor of the custom int parsing logic that was in palce before. 

### TEST PLAN

- tested locally (`/superset/queries/1588736634550.6` does not 404)
- unit tested

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by dpgaspar — COMMENTED** [review]
It would be great if you add a test that would fail before this fix and doesn't now

File: superset/views/core.py [hunk 1]
```diff
@@ -2658,14 +2658,15 @@ def fetch_datasource_metadata(self) -> FlaskResponse:
...
-    def queries(self, last_updated_ms: int) -> FlaskResponse:
+    @expose("/queries/<last_updated_ms>")
+    def queries(self, last_updated_ms: str) -> FlaskResponse:
         """
         Get the updated queries.
 
         :param last_updated_ms: unix time, milliseconds
         """
-        return self.queries_exec(last_updated_ms)
+        last_updated_ms_int = int(float(last_updated_ms)) if last_updated_ms else 0
```
Comment (villebro) [inline superset/views/core.py:2668]: I wonder if there's a cleaner way of doing `int(float(x))`

**Comment by john-bodley** [comment]
@nytai I was lead to believe that [/queries/](https://github.com/apache/incubator-superset/blob/master/superset-frontend/src/SqlLab/components/QueryAutoRefresh.jsx#L71) was always sending an integer, i.e., the timestamp rounded to the nearest millisecond. Would an alternative solution be to update the frontend logic to ensure that it sends an integer? Personally I find that cleaner especially as the backend throws away the fractional millisecond component.

**Comment by nytai** [comment]
@john-bodley We could, but since javascript doesn't really have integer support it's likely that these requests will keep happening. In either case I'm not sure 404 is what we want to do in this case, it's rather confusing from a client perspective.

**Review by willbarrett — APPROVED** [review]

**Review by craig-rueda — APPROVED** [review]

**Comment by nytai** [comment]
Merging this as is. Follow up improvements are welcome, but this issue is currently causing failures in production for us, so keeping the scope to reverting changes introduced in #9939

**Comment by ktmud** [comment]
Shouldn't the endpoint accept `float` anyway? 

```python
    @expose("/queries/<float:last_updated_ms>")
    def queries(self, last_updated_ms: float) -> FlaskResponse:
```

Fraction milliseconds seems like legit computation.

And it also kind of makes sense for `/queries/abc` to return 404...

**Comment by john-bodley** [comment]
@ktmud I agree and saw that Flask supports the `float` [url converter](https://exploreflask.com/en/latest/views.html#built-in-converters). I'll draft a PR for this.

**Comment by john-bodley** [comment]
Actually @ktmud it seems like `float` isn't valid as per [here](https://github.com/pallets/werkzeug/blob/237b35a63631fda0fe80eb51dccdfe8a2f50d304/src/werkzeug/routing.py#L1277) (and the associated regex) it _only_ supports floating point numbers whereas ideally we would want to handle both floats and integers as far as I can tell.

**Comment by ktmud** [comment]
@john-bodley Just tested locally and this works:

```diff
diff --git a/superset/views/core.py b/superset/views/core.py
index e1739721..c87c6e5b 100755
--- a/superset/views/core.py
+++ b/superset/views/core.py
@@ -2658,17 +2658,17 @@ class Superset(BaseSupersetView):
         return json_success(json.dumps(datasource.data))
 
     @has_access_api
-    @expose("/queries/<last_updated_ms>")
-    def queries(self, last_updated_ms: str) -> FlaskResponse:
+    @expose("/queries/<int:last_updated_ms>")
+    @expose("/queries/<float:last_updated_ms>")
+    def queries(self, last_updated_ms: Union[int, float]) -> FlaskResponse:
         """
         Get the updated queries.
 
         :param last_updated_ms: unix time, milliseconds
         """
-        last_updated_ms_int = int(float(last_updated_ms)) if last_updated_ms else 0
-        return self.queries_exec(last_updated_ms_int)
+        return self.queries_exec(last_updated_ms)
 
-    def queries_exec(self, last_updated_ms: int) -> FlaskResponse:
+    def queries_exec(self, last_updated_ms: Union[int, float]) -> FlaskResponse:
         stats_logger.incr("queries")
         if not g.user.get_id():
             return json_error_response(
```

**Pull request merged**